fastlane/Fastfile

```diff
@@ -433,16 +495,45 @@ platform :ios do
   #
   desc 'Trigger a new beta build on CI'
   lane :new_beta_release do |options|
-    ios_betabuild_prechecks(options)
+    # Verify that there's nothing in progress in the working copy
+    ensure_git_status_clean
+
+    # Check out the up-to-date default branch, the designated starting point for the code freeze
+    Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
+
```

Comment:
This has been removed from the WordPress iOS version, https://github.com/wordpress-mobile/WordPress-iOS/pull/21666 , on the basis that beta release are made on top of the current release branch.

Thinking about it, I think there was some logic / expectation that new betas should start only if the previous beta has been merged. I say this because I run in that scenario a number of times in WordPress and had to restart the lane passing the build code I wanted to use, which resulted in it ignoring what was on `trunk` and using `release/x.y` as the source of truth. I think the process we devised in WordPress iOS is leaner and would argue for using it here, too.

Maybe worth seeing how shipping betas for WordPress 23.6 goes?

Reply:
Yeah, I'd say let's wait and see if there are any issues with WP before we rollout that change to the other apps.

Reply:
Just a note to say that I finally deployed a beta for Jetpack and WordPress iOS and it all worked fine 😄